components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/dao/impl/AuthenticatorManagementDAOImpl.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.common.dao.impl;
+
+import org.wso2.carbon.database.utils.jdbc.NamedJdbcTemplate;
+import org.wso2.carbon.database.utils.jdbc.exceptions.TransactionException;
+import org.wso2.carbon.identity.application.common.constant.AuthenticatorMgtSQLConstants.Column;
+import org.wso2.carbon.identity.application.common.constant.AuthenticatorMgtSQLConstants.Query;
+import org.wso2.carbon.identity.application.common.dao.AuthenticatorManagementDAO;
+import org.wso2.carbon.identity.application.common.exception.AuthenticatorMgtException;
+import org.wso2.carbon.identity.application.common.exception.AuthenticatorMgtServerException;
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.application.common.model.UserDefinedLocalAuthenticatorConfig;
+import org.wso2.carbon.identity.application.common.util.AuthenticatorMgtExceptionBuilder.AuthenticatorMgtError;
+import org.wso2.carbon.identity.base.AuthenticatorPropertyConstants.AuthenticationType;
+import org.wso2.carbon.identity.base.AuthenticatorPropertyConstants.DefinedByType;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.wso2.carbon.identity.application.common.util.AuthenticatorMgtExceptionBuilder.buildServerException;
+
+/**
+ * This class implements the AuthenticatorManagementDAO interface which perform CRUD operation on database.
+ */
+public class AuthenticatorManagementDAOImpl implements AuthenticatorManagementDAO {
+
+    public static final String IS_TRUE_VALUE = "1";
+    public static final String IS_FALSE_VALUE = "0";
+    public static final String LOCAL_IDP_NAME = "LOCAL";
+
+    @Override
+    public UserDefinedLocalAuthenticatorConfig addUserDefinedLocalAuthenticator(
+            UserDefinedLocalAuthenticatorConfig authenticatorConfig, int tenantId) throws AuthenticatorMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            jdbcTemplate.withTransaction(template ->
+                jdbcTemplate.executeInsert(Query.ADD_AUTHENTICATOR_SQL,
+                    (statement -> {
```

Comment:
```suggestion
                    statement -> {
```
No need to use parenthesis.